app/importers/reading/somerville_mega_reading_importer.rb

```diff
@@ -1,67 +1,90 @@
+# For importing reading data from Google Sheets or a local folder of csvs.
 class SomervilleMegaReadingImporter
-  def initialize(options:)
-    @log = options.fetch(:log)
-    @school_local_ids = options.fetch(:school_scope, [])
-    @file_text = options.fetch(:file_text, nil)
-
-    # TODO add dependencies as instance variables
-    # @student_ids_map = ::StudentIdsMap.new
+  def initialize(school_year, options = {})
+    @school_year = school_year
+    @log = options.fetch(:log, Rails.env.test? ? LogHelper::Redirect.instance.file : STDOUT)
     # @matcher = options.fetch(:matcher, ImportMatcher.new)
-    @syncer = ::RecordSyncer.new(log: @log)
-    reset_counters!
+    # @syncer = ::RecordSyncer.new(log: @log)
+    @fetcher = options.fetch(:fetcher, GoogleSheetsFetcher.new)
   end
 
   def import
-    streaming_csv = read_or_fetch_csv
-    if streaming_csv.nil?
+    streaming_csvs = read_or_fetch_sheet
+    if streaming_csvs.nil?
       log('Aborting since no CSV found...')
       return
     end
 
-    # TODO do any initialization before the core loop
-    # log('Building student_ids_map...')
-    # @student_ids_map.reset!
-    # log("@student_ids_map built with #{@student_ids_map.size} local_id keys")
-
     log('Starting loop...')
-    reset_counters!
-    streaming_csv.each_with_index do |row, index|
-      import_row(row)
-      log("processed #{index} rows.") if index % 1000 == 0
+    @valid_student_names_count = 0
+    @invalid_student_name_count = 0
+    @valid_data_points_count = 0
+
+    processor = MegaReadingProcessor.new(read_uploaded_by_educator_from_env, @school_year)
```

Comment:
Ah cool, this is different than what I was thinking, two 😄 😄 are better than one 😄 .  This approach re-uses the process instance for each tab, which would be good for when things are stable since we can see aggregate stats across all sheets and tabs.

When deploying some other importers, I found it useful to create a new processor class for each tab, since otherwise the `stats` metrics are aggregated across all sheets and tabs, and when you see something like `invalid_student_name_count: 3` you don't know which sheet or tab to investigate or debug.  In  https://github.com/studentinsights/studentinsights/blob/master/app/importers/transitions/bedford_davis_services_importer.rb#L76 you can see how this works for example.

Let's go with this approach here and then we can see what comes up when deploying.

Reply:
Okay, I'll leave this as-is for now, but it wouldn't be much trouble to use a new instance per tab if we need to chase down problems with the sheets.